### ISSUE TITLE
Unclear match/continue documentation

### ISSUE THREAD

**Issue by paulmiller** [godotengine/godot#38770]
The [documentation](https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_basics.html#match) for `match` says, "Crash-course for people who are familiar with switch statements: …Remove any `break`s. If you don’t want to `break` by default, you can use `continue` for a fallthrough."

In other languages, "fall through" means to continue into the next switch case. For example,

```
switch(1) {
  case 1:
    std::cout << '1';
  case 2:
    std::cout << '2';
}
```

prints "12". Whereas in GDScript, `continue` means "resume matching, and only execute a subsequent block if it also matches." For example,

```
match 1:
  1:
    print(1)
    continue
  2:
    print(2)
```

prints just "1".

This tripped me up. I'd avoid comparing `continue` to "fall through", except to explicitly call out the difference.

**Comment by mrimvo** [comment]
> I'd avoid comparing continue to "fall through", except to explicitly call out the difference.

I agree on this, it's misleading. The documentation also states correctly:

> The patterns are matched from top to bottom. If a pattern matches, the corresponding block will be executed. After that, the execution continues below the match statement. If you want to have a fallthrough, you can use continue to stop execution in the current block and check the ones below it.

So, `continue` does _not_ work the same als a classic fall-through in other languages (C++, Java, ...). So this statement should probably be removed:

> If you don’t want to break by default, you can use continue for a fallthrough.

**Comment by Anutrix** [comment]
`continue` means "go check next condition"?

**Comment by theoway** [comment]
You should mention this issue here as well [.](https://github.com/godotengine/godot-docs/issues)
This is the repository for documentation on Godot's website.

**Comment by theoway** [comment]
You should mention this issue here as well.[https://github.com/godotengine/godot-docs/issues](url)
This is the repository for the documentation on Godot's website.

**Comment by paulmiller** [comment]
> So `continue` means "go check next condition"?

yup

**Comment by jitspoe** [comment]
Agreed.  I was thrown off by this as well.  Thought it was a bug.
